packages/connect-core/src/protocol-connect/create-request-header.spec.ts

```diff
@@ -0,0 +1,83 @@
+// Copyright 2021-2023 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { MethodKind } from "@bufbuild/protobuf";
+import type { Compression } from "../compression.js";
+import {
+  createRequestHeader,
+  createRequestHeaderWithCompression,
+} from "./create-request-header.js";
+import {
+  headerStreamAcceptEncoding,
+  headerStreamEncoding,
+  headerUnaryAcceptEncoding,
+  headerUnaryEncoding,
+} from "./headers.js";
+
+describe("createRequestHeader", () => {
+  it("should create request headers", () => {
+    const headers = createRequestHeader(
+      MethodKind.Unary,
+      true,
+      undefined,
+      undefined
+    );
+    expect(headers instanceof Headers).toBe(true);
+    expect(headers.get("Content-Type")).toBe("application/proto");
+    expect(headers.get("Connect-Protocol-Version")).toBe("1");
```

Comment:
Can you use [listHeaderKeys](https://github.com/bufbuild/connect-web/pull/415#discussion_r1068574063) to verify that we don't create any additional header fields? For example:

```ts
expect(listHeaderKeys(headers)).toBe(["content-type", "connect-protocol-version"]);
```

Please add it to all cases.

Reply:
sure thing